src/nvapi/nvapi_vulkan_low_latency_device.cpp

```diff
@@ -0,0 +1,225 @@
+#include "./nvapi_vulkan_low_latency_device.h"
+
+namespace dxvk {
+    std::unordered_map<VkDevice, NvapiVulkanLowLatencyDevice> NvapiVulkanLowLatencyDevice::m_lowLatencyDeviceMap = {};
+    std::mutex NvapiVulkanLowLatencyDevice::m_lowLatencyDeviceMutex = {};
+
+    std::pair<NvapiVulkanLowLatencyDevice*, VkResult> NvapiVulkanLowLatencyDevice::GetOrCreate(VkDevice device, PFN_vkGetDeviceProcAddr pvkGetDeviceProcAddr) {
+        std::scoped_lock lock{m_lowLatencyDeviceMutex};
+
+        if (auto lowLatencyDevice = Get(device))
+            return {lowLatencyDevice, VK_SUCCESS};
+
+#define VK_GET_DEVICE_PROC_ADDR(proc) auto proc = reinterpret_cast<PFN_##proc>(pvkGetDeviceProcAddr(device, #proc))
```

Comment:
Clang-tidy suggestion: Macro argument should be enclosed in parentheses 
`#define VK_GET_DEVICE_PROC_ADDR(proc) auto (proc) = reinterpret_cast<PFN_##proc>(pvkGetDeviceProcAddr(device, #proc))`

Reply:
I can do this but it feels weird. I'd never write `auto (variable) = …` myself and this is what it would expand to.

Reply:
Hmm, for me clang-tidy doesn't report this, instead it reports

```
../src/nvapi/nvml.cpp:52:89: warning: Call to virtual method 'Nvml::ErrorString' during construction bypasses virtual dispatch [clang-analyzer-optin.cplusplus.VirtualCall]
   52 |             log::info(str::format("NVML loaded but initialization failed with error: ", ErrorString(result)));
      |                                                                                         ^~~~~~~~~~~~~~~~~~~
```

and

```
../src/util/util_string.cpp:41:9: warning: Call to function 'strcpy' is insecure as it does not provide bounding of the memory buffer. Replace unbounded copy functions with analogous functions that support length arguments such as 'strlcpy'. CWE-119 [clang-analyzer-security.insecureAPI.strcpy]
   41 |         strcpy(nvss, str.c_str());
      |         ^~~~~~
```

:thinking:

Reply:
![Screenshot](https://github.com/user-attachments/assets/eed1f68a-0938-4ba9-ad8a-3b6905a10f64)
...

Interestingly I don't get your warnings, at least the first one sounds somewhat sensible..

Reply:
Perhaps clion brings its own configuration that doesn't match the default one I get when I run `clang-tidy` from CLI?

Anyway, if this is the only macro it complains about I'll just go ahead and change it.

Reply:
> Perhaps clion brings its own configuration that doesn't match the default one I get when I run `clang-tidy` from CLI?

Very likely that.
 
> I can do this but it feels weird. I'd never write `auto (variable) = …` myself and this is what it would expand to.

Actually agree, feel free to also just drop this suggestion.

Reply:
Ugh, I already committed that, then I noticed it caused the _compiler_ to complain instead:

```
../src/nvapi/nvapi_vulkan_low_latency_device.cpp:36:43: warning: unnecessary parentheses in declaration of ‘vkCreateSemaphore’ [-Wparentheses]
   36 | #define VK_GET_DEVICE_PROC_ADDR(proc) auto(proc) = reinterpret_cast<PFN_##proc>(pvkGetDeviceProcAddr(device, #proc))
      |                                           ^~~~~~
../src/nvapi/nvapi_vulkan_low_latency_device.cpp:38:9: note: in expansion of macro ‘VK_GET_DEVICE_PROC_ADDR’
   38 |         VK_GET_DEVICE_PROC_ADDR(vkCreateSemaphore);
      |         ^~~~~~~~~~~~~~~~~~~~~~~
../src/nvapi/nvapi_vulkan_low_latency_device.cpp:36:43: note: remove parentheses
   36 | #define VK_GET_DEVICE_PROC_ADDR(proc) auto(proc) = reinterpret_cast<PFN_##proc>(pvkGetDeviceProcAddr(device, #proc))
      |                                           ^~~~~~
```

I'll just drop it then. We should probably figure out some reasonable `.clang-tidy` config and commit it to the repo.